### PR TITLE
COMMERCE-9391 Product cards fragments

### PR DESCRIPTION
**Pull request by alessiorendina**
https://issues.liferay.com/browse/COMMERCE-9391

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/fragment/renderer/AvailabilityFragmentRenderer.java [hunk 1]
```diff
@@ -68,21 +72,47 @@ public void render(
...
+		if (infoItemReference != null) {
+			try {
+				ClassPKInfoItemIdentifier classPKInfoItemIdentifier =
+					(ClassPKInfoItemIdentifier)
+						infoItemReference.getInfoItemIdentifier();
+
+				cpDefinition = _cpDefinitionLocalService.getCPDefinition(
+					classPKInfoItemIdentifier.getClassPK());
+			}
+			catch (PortalException portalException) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/fragment/renderer/AvailabilityFragmentRenderer.java:90]: also the permission check

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/fragment/renderer/ImageGalleryFragmentRenderer.java [hunk 2]
```diff
@@ -68,21 +76,63 @@ public void render(
...
+
+		if (cpDefinition == null) {
+			Object infoItem = httpServletRequest.getAttribute(
+				InfoDisplayWebKeys.INFO_ITEM);
+
+			if ((infoItem == null) || !(infoItem instanceof CPDefinition)) {
+				if (_isEditMode(httpServletRequest)) {
+					_printPortletMessageInfo(
+						httpServletRequest, httpServletResponse,
+						"the-availability-component-will-be-shown-here");
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/fragment/renderer/ImageGalleryFragmentRenderer.java:125]: gallery

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/fragment/renderer/ProductCardFragmentRenderer.java [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+				ClassPKInfoItemIdentifier classPKInfoItemIdentifier =
+					(ClassPKInfoItemIdentifier)
+						infoItemReference.getInfoItemIdentifier();
+
+				cpDefinition = _cpDefinitionLocalService.getCPDefinition(
+					classPKInfoItemIdentifier.getClassPK());
+
+				if (!_commerceProductViewPermission.contains(
+						themeDisplay.getPermissionChecker(),
+						CommerceUtil.getCommerceAccountId(commerceContext),
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/fragment/renderer/ProductCardFragmentRenderer.java:148]: return in the catch block

File: modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/fragment/renderer/product_card/page.jsp [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+				<%
+				String cpDefinitionCDNURL = cpContentHelper.getCPDefinitionCDNURL(cpCatalogEntry.getCPDefinitionId(), request);
+				%>
+
+				<c:if test="<%= showImage %>">
+					<c:choose>
+						<c:when test="<%= Validator.isNotNull(cpDefinitionCDNURL) %>">
+							<img class="img-fluid product-card-picture" src="<%= cpDefinitionCDNURL %>" />
+						</c:when>
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/fragment/renderer/product_card/page.jsp:45]: move cpDefinitionCDNURL inside the "if"

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
thank you @alessiorendina

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:forward

**Pull request closed without merging**